scripts/pandas_errors_documented.py

```diff
@@ -0,0 +1,51 @@
+"""
+Check that doc/source/reference/general_utility_functions.rst documents
+all exceptions and warnings in pandas/errors/__init__.py.
+
+This is meant to be run as a pre-commit hook - to run it manually, you can do:
+
+    pre-commit run pandas-errors-documented --all-files
+"""
+from __future__ import annotations
+
+import argparse
+import ast
+import sys
+from typing import Sequence
+
+API_PATH = "doc/source/reference/general_utility_functions.rst"
```

Comment:
would this work if someone runs this hook on Windows?

Reply:
Good call!